pkg/nuctl/command/redeploy.go

```diff
@@ -0,0 +1,356 @@
+		rootCommandeer: rootCommandeer,
+		outputManifest: nuctlcommon.NewPatchManifest(),
+		betaCommandeer: betaCommandeer,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "redeploy function-name",
+		Short: "Redeploy a function",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			// initialize root
+			if err := rootCommandeer.initialize(); err != nil {
+				return errors.Wrap(err, "Failed to initialize root")
+			}
+
+			if commandeer.betaCommandeer != nil {
+				if err := commandeer.betaCommandeer.initialize(); err != nil {
+					return errors.Wrap(err, "Failed to initialize beta commandeer")
+				}
+				commandeer.rootCommandeer.loggerInstance.Debug("In BETA mode")
+				if err := commandeer.betaRedeploy(ctx, args); err != nil {
+					return errors.Wrap(err, "Failed to deploy function")
+				}
+				return nil
+			}
+			return nil
+		},
+	}
+
+	addRedeployFlags(cmd, commandeer)
+	cmd.Flags().BoolVarP(&commandeer.skipSpecCleanup, "skip-spec-cleanup", "", false, "Do not clean up spec in function configs")
+
+	commandeer.cmd = cmd
+
+	return commandeer
+}
+
+func addRedeployFlags(cmd *cobra.Command,
+	commandeer *redeployCommandeer) {
+	cmd.Flags().BoolVar(&commandeer.deployAll, "deploy-all", false, "Deploy all functions in the namespace")
+	cmd.PersistentFlags().BoolVarP(&commandeer.waitForFunction, "wait", "w", false, "Wait for function deployment to complete")
+	cmd.PersistentFlags().StringSliceVar(&commandeer.excludedProjects, "exclude-projects", []string{}, "Exclude projects to patch")
+	cmd.PersistentFlags().StringSliceVar(&commandeer.excludedFunctions, "exclude-functions", []string{}, "Exclude functions to patch")
+	cmd.PersistentFlags().BoolVar(&commandeer.excludeFunctionWithGPU, "exclude-functions-with-gpu", false, "Skip functions with GPU")
+	cmd.PersistentFlags().BoolVar(&commandeer.importedOnly, "imported-only", false, "Deploy only imported functions")
+	cmd.PersistentFlags().DurationVar(&commandeer.waitTimeout, "wait-timeout", 15*time.Minute, "Wait timeout duration for the function deployment (default 15m)")
+	cmd.Flags().BoolVarP(&commandeer.redeployFromReportFile, "redeploy-from-report-file", "", false, "Redeploy failed and retryable functions from the given report file")
+	cmd.Flags().BoolVarP(&commandeer.saveReport, "save-report", "", false, "Save redeployment report to a file")
+	cmd.Flags().BoolVarP(&commandeer.verifyExternalRegistry, "verify-external-registry", "", false, "verify registry is external")
+	cmd.Flags().StringVarP(&commandeer.reportFilePath, "report-file-path", "", "nuctl-redeployment-report.json", "Path to redeployment report")
+}
+
+func (d *redeployCommandeer) betaRedeploy(ctx context.Context, args []string) error {
+
+	if d.redeployFromReportFile {
+		manifest, err := nuctlcommon.NewPatchManifestFromFile(d.reportFilePath)
+		if err != nil {
+			return errors.Wrap(err, "Problem with reading report file")
+		}
+		d.inputManifest = manifest
+		retryableFunctions := d.inputManifest.GetRetryableFunctionNames()
```

Comment:
We can skip the middleman and remove `d.inputManifest` completely as it is never used later
```suggestion
		retryableFunctions := manifest.GetRetryableFunctionNames()
```